zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/auth/QuorumAuthTestBase.java

```diff
@@ -44,11 +44,11 @@ public class QuorumAuthTestBase extends ZKTestCase {
 
     protected static final Logger LOG = LoggerFactory.getLogger(QuorumAuthTestBase.class);
     protected List<MainThread> mt = new ArrayList<>();
+    @TempDir
     protected static File jaasConfigDir;
```

Comment:
Thats a great, question. I dont know.. This will need some investigation. I will look into this. Meantime I am going to revert this and avoid using the annotation only for this. But its a good deep dive I think..

Reply:
Should avoid static initializer blocks in general... they make for hard-to-test code, not to mention the potential bootstrapping issue mentioned here. Should use JUnit `@BeforeAll` annotation to initialize static fields in test case classes, if possible, to make sure JUnit does the right thing automatically.

Reply:
I created this https://issues.apache.org/jira/browse/ZOOKEEPER-4789 to work on cleaning the QuorumAuth tests.